internal/retry/retry.go

```diff
@@ -0,0 +1,158 @@
+package retry
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/pkg/errors"
+
+	"github.com/dapr/components-contrib/internal/config"
+)
+
+// PolicyType denotes if the back off delay should be constant or exponential.
+type PolicyType int
+
+const (
+	// PolicyConstant is a backoff policy that always returns the same backoff delay.
+	PolicyConstant PolicyType = iota
```

Comment:
What is this `itoa` trick?

Reply:
https://yourbasic.org/golang/iota/
https://golang.org/ref/spec#Iota

Reply:
LGTM! I learned something new.